packages/plugin-lighthouse/src/lib/utils.ts

```diff
@@ -68,3 +73,64 @@ export function validateOnlyAudits(
   }
   return true;
 }
+
+export class CategoriesNotImplementedError extends Error {
+  constructor(categorySlugs: string[]) {
+    super(`categories: "${categorySlugs.join(', ')}" not implemented`);
+  }
+}
+
+export function validateOnlyCategories(
+  groups: Group[],
+  onlyCategories: string | string[],
+): boolean {
+  const missingCategories = toArray(onlyCategories).filter(
+    slug => !groups.some(group => group.slug === slug),
+  );
+  if (missingCategories.length > 0) {
+    throw new CategoriesNotImplementedError(missingCategories);
+  }
+  return true;
+}
+
+export function filterAuditsAndGroupsByOnlyOptions(
+  audits: Audit[],
+  groups: Group[],
+  options?: Pick<LighthouseFlags, 'onlyAudits' | 'onlyCategories'>,
+): {
+  audits: Audit[];
+  groups: Group[];
+} {
+  const { onlyAudits, onlyCategories } = options ?? {};
+
+  // category wins over audits
+  if (onlyCategories && onlyCategories.length > 0) {
+    validateOnlyCategories(groups, onlyCategories);
+
+    const categorieSlugs = new Set(onlyCategories);
+    const filteredGroups: Group[] = filterBy<Group>(groups, ({ slug }) =>
+      categorieSlugs.has(slug),
+    );
```

Comment:
The `Group` type is inferred from the `groups` parameter:

```suggestion
    const filteredGroups = filterBy(groups, ({ slug }) =>
      categorieSlugs.has(slug),
    );
```

Reply:
true